scheduler/testing.go

```diff
@@ -262,7 +262,7 @@ func (h *Harness) Snapshot() State {
 // a snapshot of current state using the harness for planning.
 func (h *Harness) Scheduler(factory Factory) Scheduler {
 	logger := testlog.HCLogger(h.t)
-	return factory(logger, h.Snapshot(), h)
+	return factory(logger, nil, h.Snapshot(), h)
```

Comment:
Not a requirement but it might be interesting to add an implementation here to fail the test if an invariant is broken.

I bet we test against bad data all the time because the test just doesn't happen to exercise it or care. It might be pretty annoying to fix but if it yields even a single lead toward somewhere we could legitimately end up with bad data it seems worth it.

Reply:
Since I don't have any tests where we would expect an event here, I added some logic to fail the test if something ever shows up here.